exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -0,0 +1,218 @@
+    buckets = []
+    total_count = 0
+    for index, value in enumerate(metric.point.bucket_counts):
+        total_count += value
+        buckets.append(
+            (
+                f"{metric.point.explicit_bounds[index]}",
+                total_count,
+            )
+        )
+    return buckets
+
+
+class PrometheusMetricReader(MetricReader):
+    """Prometheus metric exporter for OpenTelemetry.
+
+    Args:
+        prefix: single-word application prefix relevant to the domain
+            the metric belongs to.
+    """
+
+    def __init__(self, prefix: str = "") -> None:
+        super().__init__()
+        self._collector = _CustomCollector(self.collect, prefix)
+        core.REGISTRY.register(self._collector)
+
+    def _receive_metrics(self, metrics: Iterable[Metric]) -> None:
+        if metrics is None:
+            return
+        token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        try:
+            self._collector.add_metrics_data(metrics)
+        except Exception as e:  # pylint: disable=broad-except,invalid-name
+            _logger.exception("Exception while exporting metrics %s", str(e))
+        detach(token)
+
+    def shutdown(self) -> bool:
+        core.REGISTRY.unregister(self._collector)
+        return True
+
+
+class _CustomCollector:
+    """_CustomCollector represents the Prometheus Collector object
+
+    See more:
+    https://github.com/prometheus/client_python#custom-collectors
+    """
+
+    def __init__(self, callback, prefix: str = ""):
+        self._prefix = prefix
+        self._callback = callback
+        self._metrics_to_export = collections.deque()
+        self._non_letters_digits_underscore_re = re.compile(
+            r"[^\w]", re.UNICODE | re.IGNORECASE
+        )
+
+    def add_metrics_data(self, export_records: Sequence[Metric]) -> None:
+        """Add metrics to Prometheus data"""
+        self._metrics_to_export.append(export_records)
+
+    def collect(self) -> None:
+        """Collect fetches the metrics from OpenTelemetry
+        and delivers them as Prometheus Metrics.
+        Collect is invoked every time a ``prometheus.Gatherer`` is run
+        for example when the HTTP endpoint is invoked by Prometheus.
+        """
+        if self._callback is not None:
+            self._callback()
+
+        while self._metrics_to_export:
+            for export_record in self._metrics_to_export.popleft():
+                prometheus_metric = self._translate_to_prometheus(
+                    export_record
+                )
+                if prometheus_metric is not None:
+                    yield prometheus_metric
+
+    def _translate_to_prometheus(
+        self, metric: Metric
+    ) -> Optional[core.Metric]:
+        prometheus_metric = None
+        label_values = []
+        label_keys = []
+        for key, value in metric.attributes.items():
+            label_keys.append(self._sanitize(key))
+            label_values.append(str(value))
```

Comment:
What happens to list? Is it allowed?

Reply:
Here is some relevant discussion https://github.com/open-telemetry/opentelemetry-specification/pull/2266#discussion_r804900011. I didn't read the linked spec thoroughly but I think we should make this complaint to it. ^spec [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#otlp-metric-points-to-prometheus).

Reply:
Based on the attribute specification, it looks like the list should just be converted to its JSON representation. Are you ok with me opening a follow up issue for it?

Reply:
Sure

Reply:
#2502